src/eckit/codec/detail/TypeTraits.h

```diff
@@ -14,7 +14,9 @@
 
 #include <type_traits>
 
-#include "eckit/codec/detail/DataType.h"
+#include "eckit/codec/detail/DataType.h"  // enforce visibility
+#include "eckit/codec/types/scalar.h"     // ...
+#include "eckit/codec/types/string.h"     // ...
```

Comment:
I would not add scalar or string here in TypeTraits class.
You could add it just where needed.
Anyway it is already exported in "eckit/codec/codec.h"